### PR TITLE
Swift 3.0 Support

### PR DESCRIPTION
**Pull request by adolfo**
* Adds support for Swift 3.0
* Updates Travis build config for Xcode 8
* Bumps version and podspec to 1.1.0
* Updates README with Swift 3.0 examples
* Updates tests

### CODE REVIEW

File: SwiftMQTT.podspec [hunk 1]
```diff
@@ -2,10 +2,10 @@
 Pod::Spec.new do |s|
     
     s.name         = "SwiftMQTT"
-    s.version      = "1.0.2"
-    s.summary      = "MQTT Client in pure swift"
+    s.version      = "1.1.0"
```
Comment (bhargavg) [inline SwiftMQTT.podspec]: I feel this can be "2.0.0" as we've changed the public API. What do you think?

**Comment by adolfo** [comment]
Yes, good idea, we should bump the major version. You think we should revisit the public methods to be more concise and "swift-3.0y" now that we're making breaking changes?

**Comment by bhargavg** [comment]
Yes, that would be great.

**Comment by adolfo** [comment]
@bhargavg Not sure why CI fails sometimes and succeeds other times...
Take a look at last two commits – feedback welcome.

**Comment by bhargavg** [comment]
@adolfo Awesome work.

Regarding the travis build failure, I'm getting the same random failures on my projects too. Not sure what is causing this.

I've restarted the build and it worked again  ¯\_(ツ)_/¯

**Pull request merged**